toolchain/sem_ir/constant.cpp

```diff
@@ -9,6 +9,16 @@
 namespace Carbon::SemIR {
 
 auto ConstantStore::GetOrAdd(Inst inst, bool is_symbolic) -> ConstantId {
+  // Check that we're allowed to form this kind of constant.
+  CARBON_CHECK(inst.kind().constant_kind() != InstConstantKind::Never)
+      << "Should not form a constant from instruction " << inst;
+  CARBON_CHECK(inst.kind().constant_kind() != InstConstantKind::SymbolicOnly ||
+               is_symbolic)
+      << "Should only form a symbolic constant from instruction " << inst;
+  CARBON_CHECK(inst.kind().constant_kind() != InstConstantKind::Always ||
+               !is_symbolic)
+      << "Should only form a template constant from instruction " << inst;
```

Comment:
Can this be written differently so that either one CHECK runs or something? My mind just doesn't operate straight on this kind of "!=" code structure. A `switch` might make this easier to reason about.

Reply:
Done.